src/store.ts

```diff
@@ -1,30 +1,99 @@
 import Vue from 'vue';
-import Vuex from 'vuex';
-import firebase from 'firebase/app';
+import Vuex, { Store } from 'vuex';
 
 import * as fb from './firebaseConfig';
+import { AppToggleableRequirementChoices } from './user-data';
+import { checkNotNull } from './utilities';
 
 Vue.use(Vuex);
 
-type State = {
-  currentUser: firebase.User | null;
+type SimplifiedFirebaseUser = { readonly displayName: string; readonly email: string };
+
+export type VuexStoreState = {
+  currentFirebaseUser: SimplifiedFirebaseUser;
+  toggleableRequirementChoices: AppToggleableRequirementChoices;
+  subjectColors: Readonly<Record<string, string>>;
+  uniqueIncrementer: number;
 };
 
-const store = new Vuex.Store<State>({
+export class TypedVuexStore extends Store<VuexStoreState> {}
+
+const store: TypedVuexStore = new TypedVuexStore({
   state: {
-    currentUser: null,
+    // We allow the initial value to be null, so that the value can be read non-null elsewhere.
+    // It is only null when the application has not authenticated the user yet, which is a scanario
+    // only need to be considered in Login component.
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    currentFirebaseUser: null!,
+    toggleableRequirementChoices: {},
+    subjectColors: {},
+    uniqueIncrementer: 0,
   },
   actions: {},
   mutations: {
-    setCurrentUser(state: State, val: firebase.User) {
-      state.currentUser = val;
+    setCurrentFirebaseUser(state: VuexStoreState, user: SimplifiedFirebaseUser) {
+      state.currentFirebaseUser = user;
+    },
+    setToggleableRequirementChoices(
+      state: VuexStoreState,
+      toggleableRequirementChoices: AppToggleableRequirementChoices
+    ) {
+      state.toggleableRequirementChoices = toggleableRequirementChoices;
+    },
+    setSubjectColors(state: VuexStoreState, colors: Readonly<Record<string, string>>) {
+      state.subjectColors = colors;
+    },
+    setUniqueIncrementer(state: VuexStoreState, newIncrementerValue: number) {
+      state.uniqueIncrementer = newIncrementerValue;
     },
   },
 });
 
+export const subscribeToggleableRequirementChoicesChange = (
+  handler: (state: VuexStoreState) => void
+): (() => void) =>
+  store.subscribe((payload, state) => {
+    if (payload.type === 'setToggleableRequirementChoices') {
+      handler(state);
+    }
+  });
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+let firestoreListenerUnsubscriber = (): void => {};
+
 fb.auth.onAuthStateChanged(user => {
   if (user) {
-    store.commit('setCurrentUser', user);
+    const simplifiedUser = {
+      displayName: checkNotNull(user.displayName),
+      email: checkNotNull(user.email),
+    };
```

Comment:
Enforce non-null display name and email here once and for all, instead of checking it everywhere else